packages/@azure/servicebus/data-plane/test/sessionsTests.spec.ts

```diff
@@ -0,0 +1,494 @@
+    void
+  > {
+    await test_streaming(unpartitionedTopicSessionClient, unpartitionedSubscriptionSessionClient);
+  });
+});
+
+describe("Accept a session without passing sessionId and receive messages - Queue/Subscription has different messages belonging to different sessionId", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function testComplete_batching(
+    senderClient: QueueClient | TopicClient,
+    sessionClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+
+    await senderClient.send(testMessagesWithDifferentSessionIds[0]);
+    await senderClient.send(testMessagesWithDifferentSessionIds[1]);
+    await delay(4000);
+
+    let receiverClient = await sessionClient.acceptSession();
+    let msgs = await receiverClient.receiveBatch(2);
+
+    should.equal(Array.isArray(msgs), true);
+    should.equal(msgs.length, 1);
+    
+    should.equal(
+      testMessagesWithDifferentSessionIds.some((x) => msgs[0].body === x.body && msgs[0].messageId === x.messageId && msgs[0].sessionId === x.sessionId),
+      true,
+      "Received Message doesnt match any of the test messages"
+    );
+    await msgs[0].complete();
+    
+    receiverClient = await sessionClient.acceptSession();
+    await testPeekMsgsLength(receiverClient, 1);
+    msgs = await receiverClient.receiveBatch(2);
+
+    should.equal(Array.isArray(msgs), true);
+    should.equal(msgs.length, 1);
+    should.equal(
+      testMessagesWithDifferentSessionIds.some((x) => msgs[0].body === x.body && msgs[0].messageId === x.messageId && msgs[0].sessionId === x.sessionId),
+      true,
+      "Received Message doesnt match any of the test messages"
+    );
+    await msgs[0].complete();
+    await testPeekMsgsLength(receiverClient, 0);
+  }
+
+
+  it("Partitioned Queues with Sessions - Batch Receiver: complete() removes message", async function(): Promise<
+    void
+  > {
+    await testComplete_batching(partitionedQueueSessionClient, partitionedQueueSessionClient);
+  });
+
+  it("Partitioned Topics and Subscription with Sessions - Batch Receiver: complete() removes message", async function(): Promise<
+    void
+  > {
+    await testComplete_batching(partitionedTopicSessionClient, partitionedSubscriptionSessionClient);
+  });
+
+  it("Unpartitioned Queues with Sessions - Batch Receiver: complete() removes message", async function(): Promise<
+    void
+  > {
+    await testComplete_batching(unpartitionedQueueSessionClient, unpartitionedQueueSessionClient);
+  });
+
+  it("Unpartitioned Topics and Subscription with Sessions - Batch Receiver: complete() removes message", async function(): Promise<
+    void
+  > {
+    await testComplete_batching(unpartitionedTopicSessionClient, unpartitionedSubscriptionSessionClient);
+  });
+
+  async function testComplete_streaming(
```

Comment:
We dont need the streaming variation here.
What we are testing is that `acceptSession()` randomly picks one session, and the second time it picks another one. The batching set is enough

Reply:
Makes sense. Removed the streaming tests for this.